src/sentry_sync.h

```diff
@@ -324,6 +324,12 @@ int sentry__bgworker_start(sentry_bgworker_t *bgw);
  */
 int sentry__bgworker_shutdown(sentry_bgworker_t *bgw, uint64_t timeout);
 
+/**
+ * This will set a preferable thread name for background worker.
+ * Should be executed before worker start
+ */
+void sentry__bgworker_setname(sentry_bgworker_t *bgw, const char *thread_name);
```

Comment:
maybe instead of an explicit method, make this a new parameter of `sentry__bgworker_new`

Reply:
I tried, in this case it becomes even more complicated.
 
For example, WinHTTP transport creates bgworker in `sentry__transport_new_default()` which called by `sentry_options_new()`. With this approach we have no ability to set name via options_set_... because bgworker was already created.